### PR TITLE
Feature/task manager test

### PR DESCRIPTION
**Pull request by Krisha434**
Add tests to the task manager.

### CODE REVIEW

File: dbtest.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    """Sets up an in-memory SQLite database for testing."""
+    conn = sqlite3.connect(":memory:")
+    cursor = conn.cursor()
+    cursor.execute(CREATE_TASKS_TABLE)
+    conn.commit()
+    yield conn
+    conn.close()
+
+@pytest.fixture
+def app_with_test_db(monkeypatch):
```
Comment (sreedevijain19) [inline dbtest.py:24]: Ensure that the table creation (create_tasks_table()) is handled in the test_db fixture itself, Can remove the redundant table creation in the app_with_test_db fixture and rely on the test_db fixture to handle database setup.

File: dbtest.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    """Mocks the database connection to use the in-memory test database."""
```
Comment (sreedevijain19) [inline dbtest.py:25]: The table creation part inside this is unnecessary because you already create the table in the test_db fixture. Just mock the connection part in the app_with_test_db fixture and ensure create_tasks_table() is not called here.

File: dbtest.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import sqlite3
+import os
+from dotenv import load_dotenv
+from database import get_connection, create_tasks_table
+from queries import CREATE_TASKS_TABLE, INSERT_TASK, GET_ALL_TASKS, UPDATE_TASK, DELETE_TASK
+from main import add_task, get_all_tasks, update_task, delete_task, validate_priority
+
+# Load environment variables for testing
+load_dotenv()
+TEST_DB_PATH = os.getenv("DB_PATH", "task_manager.db")
```
Comment (sreedevijain19) [inline dbtest.py:11]: consider adding check to get this env.

File: dbtest.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+# --- Tests for queries.py ---
+
+def test_query_definitions():
+    """Ensures that the query constants are defined as strings."""
+    assert isinstance(CREATE_TASKS_TABLE, str)
+    assert isinstance(INSERT_TASK, str)
+    assert isinstance(GET_ALL_TASKS, str)
+    assert isinstance(UPDATE_TASK, str)
+    assert isinstance(DELETE_TASK, str)
+    assert "tasks" in CREATE_TASKS_TABLE
```
Comment (sreedevijain19) [inline dbtest.py:62]: Instead of just checking if "tasks" is in the query, also verify the query against actual SQL syntax to ensure it's correct and doesn't contain potential errors (e.g., missing commas, incorrect columns).

File: dbtest.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    conn = get_connection()
+    assert isinstance(conn, sqlite3.Connection)
+    conn.close()
+
+def test_get_connection_missing_path(monkeypatch):
+    """Tests that get_connection raises ValueError if DB_PATH is not set."""
+    monkeypatch.delenv("DB_PATH", raising=False)
+    with pytest.raises(ValueError) as excinfo:
+        get_connection()
+    assert "Database path not found in environment variables (.env)" in str(excinfo.value)
```
Comment (sreedevijain19) [inline dbtest.py:43]: ensure that the exception message is specific and points to the exact issue (e.g., "DB_PATH environment variable is not set")

File: dbtest.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    validate_priority("High")
+
+def test_validate_priority_invalid():
+    """Tests that validate_priority raises ValueError for invalid priorities."""
+    with pytest.raises(ValueError) as excinfo:
+        validate_priority("Urgent")
+    assert "Priority must be one of: Low, Medium, High" in str(excinfo.value)
+
+def test_add_task(app_with_test_db):
+    """Tests that add_task successfully inserts a task into the database."""
```
Comment (sreedevijain19) [inline dbtest.py:81]: Test for edge cases like adding a task with a missing or invalid priority.

File: dbtest.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    with get_connection() as conn:
+        cursor = conn.cursor()
+        cursor.execute("SELECT title, description, priority, due_date FROM tasks WHERE title = ?", ("Buy groceries",))
+        task = cursor.fetchone()
+        assert task is not None
+        assert task == ("Buy groceries", "Milk, eggs, bread", "Medium", "2025-05-10")
+
+def test_update_task(app_with_test_db):
+    """Tests that update_task successfully modifies an existing task."""
+    with get_connection() as conn:
```
Comment (sreedevijain19) [inline dbtest.py:92]: Test for invalid task IDs (non-existing task IDs).

File: dbtest.py [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    update_task(task_id, "New Title", "New Desc", "High", "2025-05-10", 1)
+
+    with get_connection() as conn:
+        cursor = conn.cursor()
+        cursor.execute("SELECT title, description, priority, due_date, completed FROM tasks WHERE id = ?", (task_id,))
+        updated_task = cursor.fetchone()
+        assert updated_task == ("New Title", "New Desc", "High", "2025-05-10", 1)
+
+def test_delete_task(app_with_test_db):
+    """Tests that delete_task successfully removes a task from the database."""
```
Comment (sreedevijain19) [inline dbtest.py:107]: Try deleting a task that doesn't exist and ensure it raises the correct exception
